lib/article_json/import/google_doc/html/embedded_slideshare_element.rb

```diff
@@ -0,0 +1,35 @@
+module ArticleJSON
+  module Import
+    module GoogleDoc
+      module HTML
+        class EmbeddedSlideshareElement < EmbeddedElement
+          # The type of this embedded element
+          # @return [Symbol]
+          def embed_type
+            :slideshare
+          end
+
+          # Extract the slide show ID (including the handle) from an URL
+          # @return [String]
+          def embed_id
+            match = @node.inner_text.strip.match(self.class.url_regexp)
+            "#{match[:handle]}/#{match[:id]}" if match
+          end
+
+          class << self
+            # Regular expression to check if a given string is a Slideshare URL
+            # Also used to extract HANDLE and ID from the URL.
+            # @return [Regexp]
+            def url_regexp
```

Comment:
just curious, why do you need this method to be a class method?

Reply:
because another class method is calling it (in the base class):
https://github.com/Devex/article_json/blob/master/lib/article_json/import/google_doc/html/embedded_element.rb#L62

Reply:
I see, thanks!